### PR TITLE
fruit: Conan V2 and add v3.7.0

### PR DESCRIPTION
**Pull request by tiolan**
Specify library name and version:  fruit/3.7.0

fruit 3.7.0 was recently released, so I add it to the Conan Center
https://github.com/google/fruit/releases/tag/v3.7.0

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- Conan version 1.55.0
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.
- No errors
- Warnings as before

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]
LGTM

**Comment by tiolan** [comment]
The build fails for this config:
`Macos | armv8 | apple-clang | 13.0`
with `fatal error: 'boost/unordered_map.hpp' file not found`. Probably a boost-package issue.
I tried to reproduce locally on Ubuntu amd64 with clang 13, but it worked.
All possible combinations with gcc11, gcc12, clang13 worked for me.
I do not have access to any apple-clang armv8 compiler in order to reproduce.
@uilianries @tt4g Any idea how to continue? Would disabling `use_boost` for apple-clang be an option? Likely not.

**Comment by tt4g** [comment]
@tiolan Unfortunately, it is the exact same problem as my PR (#14569).
I have not been able to reproduce this problem either, so I suspect it is a CI-specific glitch.

The `use_boost=True` is specified because it was [the default behavior for fruit](https://github.com/google/fruit/blob/master/CMakeLists.txt#L71-L73). You can also set `use_boost=False`, but CI is supposed to test all patterns, so you will encounter the same error.

**Comment by tiolan** [comment]
@tt4g 
Using some hacks (see below) I was able to reproduce locally with my Ubuntu Clang.
It turned out boost was pulled as `build_requirements` but not as a `requirements` and so when cross-compiling, it was not included in `conanbuildinfo.txt`, causing `CONAN_INCLUDE_DIRS_BOOST` not to be defined in the patched `CMakeLists.txt`.
I guess in future the patches should be cleaned up to use `find_package(Boost REQUIRED)` instead of adding include-dirs based on variables defined by Conan. But this is a larger change.

For reference:
This will obviously fail at link stage, but compiles under Ubuntu
`CC=clang-13 CXX=clang++-13 conan create recipes/fruit/all/ fruit/3.7.0@ -pr:h apple-host -pr:b apple-build`
comment out `conan_basic_setup()`
and build with profile apple-build
```
[settings]
arch=x86_64
build_type=Debug
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13.0
os=Macos
[options]
fruit:shared=True
[build_requires]
[env]
[conf]
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True
tools.apple:sdk_path=<path/to/>/MacOSX-SDKs/MacOSX11.3.sdk
```

and profile apple-host
```
[settings]
arch=armv8
build_type=Debug
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13.0
os=Macos
[options]
fruit:shared=True
[build_requires]
[env]
[conf]
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True
tools.apple:sdk_path=<path/to/>/MacOSX-SDKs/MacOSX11.3.sdk
```

having https://github.com/phracker/MacOSX-SDKs cloned to `<path/to/>/MacOSX-SDKs`

@uilianries ready for review.

**Comment by uilianries** [comment]
@tiolan I would suggest trying new Conan generators, like CMakeDeps and try it again. We are adopting CMake Targets by default, so CMake variables are no longer the main option. I gonna send you a PR updating some parts, so you can check without those new patches.

**Comment by uilianries** [comment]
@tiolan Please, consider https://github.com/tiolan/conan-center-index/pull/2

File: recipes/fruit/all/conanfile.py
Comment (tiolan) [inline recipes/fruit/all/conanfile.py:92]: @uilianries Do we need `BUILD_SHARED_LIBS=False`, if `self.options.shared==False` here? Or does CMakeToolchain it automatically?

Comment (uilianries) [reply recipes/fruit/all/conanfile.py:92]: @tiolan It's automatically managed: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#extending-and-customizing-cmaketoolchain

File: recipes/fruit/all/conanfile.py
Comment (tiolan) [inline recipes/fruit/all/conanfile.py:40]: @uilianries Was there a specific reason for you to update to newest boost?
The pipeline did not find the boost package. I do not yet know why.
When searching manually, I find it.

`conan search 'boost/1.80.0@' -r conancenter -q 'arch=x86_64 AND build_type=Release AND compiler=gcc AND compiler.libcxx=libstdc++11 AND compiler.version=11 AND os=Linux AND shared=False'`

**Comment by tiolan** [comment]
@uilianries ready for review.

File: recipes/fruit/all/conanfile.py
Comment (prince-chrismc) [reply recipes/fruit/all/conanfile.py:44]: gcc 11 might have been missing but this seems to be passing now

File: recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch
Comment (prince-chrismc) [inline recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch:27]: Why do we need play with the warnings? does this fail the build?

**Review by prince-chrismc — COMMENTED** [review]
Just one question but overall this looks super good

File: recipes/fruit/all/conanfile.py
Comment (StellaSmith) [inline recipes/fruit/all/conanfile.py:26]: using `with_boost` sounds more consistent with the rest of the recipes than `use_boost`

File: recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch
Comment (tt4g) [reply recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch:27]: I will answer since I was the first to make this patch.
This flag was set in the original fruit repository. I don't know why this flag is needed.
Nevertheless, removing the warnings may make it more difficult to investigate defects because the warnings that library developers expect to be displayed at build time will no longer be displayed, so I decided to keep the flags.

TIP: The original `CMakeLists.txt` did not support CMake multi-configuration-generator, so it has been modified.

File: recipes/fruit/all/conanfile.py
Comment (prince-chrismc) [reply recipes/fruit/all/conanfile.py:26]: 👍  on this good catch!

File: recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch
Comment (prince-chrismc) [reply recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch:27]: Ahh then i'd highly recommend making a PR against upstream to contribute back this improvement  :)

Adding support for multi-config generators is fine :)

File: recipes/fruit/all/conanfile.py
Comment (tiolan) [reply recipes/fruit/all/conanfile.py:26]: True, but it had always been like this.
And I guess fruit in their CMake call it like this.
So do we change it? Would break how people are used to use this package.

File: recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch
Comment (tt4g) [reply recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch:27]: I remember not making PR because the maintainer was busy at the time and did not seem to have time to maintain it.
The maintainer's response has been quick, so they may be able to check PR now.

I am currently away from the C/C++ project and don't have much time available, so it would be great if someone could create PR.

File: recipes/fruit/all/conanfile.py
Comment (prince-chrismc) [reply recipes/fruit/all/conanfile.py:26]: We can [deprecate the option](https://github.com/conan-io/conan-center-index/blob/c1361ec2150a7dcb966608c39b6c12bc34e07858/docs/faqs.md?plain=1#L166) and read it incase it's used  to help with migration

No this is not required but ideally

File: recipes/fruit/all/conandata.yml
Comment (prince-chrismc) [inline recipes/fruit/all/conandata.yml:11]: Ideally new version are on top (thats the order the build service works which leads to faster results)

File: recipes/fruit/all/conanfile.py
Comment (prince-chrismc) [inline recipes/fruit/all/conanfile.py:37]:
```suggestion
            self.options.rm_safe("fPIC")
```

**Review by prince-chrismc — APPROVED** [review]

File: recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch
Comment (tt4g) [reply recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch:27]: Open PR now: google/fruit#149

File: recipes/fruit/all/conandata.yml
Comment (tiolan) [reply recipes/fruit/all/conandata.yml:11]: I would propose to close this PR with this change the way it is and the  when I add fruit 3.7.1, I will adapt accordingly. Agreed?

File: recipes/fruit/all/conanfile.py
Comment (tiolan) [reply recipes/fruit/all/conanfile.py:26]: I would propose to close this PR with this change the way it is and the  when I add fruit 3.7.1, I will adapt accordingly. Agreed?

File: recipes/fruit/all/conandata.yml
Comment (prince-chrismc) [reply recipes/fruit/all/conandata.yml:11]: Absolutely, not a blocker just an FYI ❤️ just for whenever the next PR is, it's not worth one on it's own

File: recipes/fruit/all/conanfile.py
Comment (RubenRBS) [inline recipes/fruit/all/conanfile.py:97]: Sorry if this is a dumb question, but can you tell me why this definition was left out of the v2 conversion? I'm guessing it does not make sense to have it anymore? Or is it releated to https://github.com/conan-io/conan-center-index/pull/15494/files#diff-9a0d263d27f331b7f71ed0a2f4e39864d4e1ba6df602130a0094d84b52109612R434?

**Review by RubenRBS — COMMENTED** [review]
Hi! Thanks for your work, we really appreciate it, sorry that this is took so long to get reviewed, we're hard at work for the 2.0 launch, and we're trying to catch up after the holidays still :)

This looks good to me, but before I accept I have a minor question, so once that's answered, let's get this merged :)

File: recipes/fruit/all/conanfile.py
Comment (uilianries) [reply recipes/fruit/all/conanfile.py:97]: It's False already by default: https://github.com/google/fruit/blob/v3.7.0/CMakeLists.txt

**Review by uilianries — APPROVED** [review]

**Pull request merged**
